src/core/openthread-core-default-config.h

```diff
@@ -753,13 +753,16 @@
 #endif
 
 /**
- * @def OPENTHREAD_CONFIG_ENABLE_BEACON_RSP_IF_JOINABLE
+ * @def OPENTHREAD_CONFIG_ENABLE_BEACON_RSP_WHEN_JOINABLE
  *
- * Define to 1 if you want to enable beacon response for joinable networks.
+ * Define to 1 if you want to enable beacon response device is made joinable.
+ *
+ * Note that when this feature is enabled, the device will respond to beacon message when it is made joinable EVEN if
+ * it is in child or detached modes.
```

Comment:
suggestion: `@note  When this feature is enabled, the device will transmit IEEE 802.15.4 Beacons in response to IEEE 802.15.4 Beacon Requests even while the device is not router capable and detached.`

Reply:
Done. Thank you :)
Also updated the commit log...